File: java/08_050.java
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    
    TreeNode(int x) {
        val = x;
    }
}

public class Solution {
    public TreeNode symmetrizeTree(TreeNode root) {
        if (root == null) {
            return null;
        }
        
        // Transform left and right subtrees first
        root.left = symmetrizeTree(root.left);
        root.right = symmetrizeTree(root.right);
        
        // Check if subtrees are symmetric
        if (!isSymmetric(root.left, root.right)) {
            // Not symmetric, make them symmetric
            if (root.left != null && root.right == null) {
                // Only left child exists, mirror it
                root.right = mirrorTree(root.left);
            } else if (root.right != null && root.left == null) {
                // Only right child exists, mirror it
                root.left = mirrorTree(root.right);
            } else {
                // Both children exist but not symmetric
                // Choose left subtree as reference and mirror it
                root.right = mirrorTree(root.left);
            }
        }
        
        return root;
    }
    
    private boolean isSymmetric(TreeNode left, TreeNode right) {
        if (left == null && right == null) {
            return true;
        }
        if (left == null || right == null) {
            return false;
        }
        if (left.val != right.val) {
            return false;
        }
        
        return isSymmetric(left.left, right.right) && 
               isSymmetric(left.right, right.left);
    }
    
    private TreeNode mirrorTree(TreeNode node) {
        if (node == null) {
            return null;
        }
        
        TreeNode mirrored = new TreeNode(node.val);
        mirrored.left = mirrorTree(node.right);
        mirrored.right = mirrorTree(node.left);
        
        return mirrored;
    }
}

File: python/08_050.py
class TreeNode:

    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def symmetrize_tree(root):
    if not root:
        return None
    
    # Transform left and right subtrees first
    root.left = symmetrize_tree(root.left)
    root.right = symmetrize_tree(root.right)
    
    # Check if subtrees are symmetric
    if not is_symmetric(root.left, root.right):
        # Not symmetric, make them symmetric
        if root.left and not root.right:
            # Only left child exists, mirror it
            root.right = mirror_tree(root.left)
        elif root.right and not root.left:
            # Only right child exists, mirror it
            root.left = mirror_tree(root.right)
        else:
            # Both children exist but not symmetric
            # Choose left subtree as reference and mirror it
            root.right = mirror_tree(root.left)
    
    return root

def is_symmetric(left, right):
    if not left and not right:
        return True
    if not left or not right:
        return False
    if left.val != right.val:
        return False
    
    return (is_symmetric(left.left, right.right) and
            is_symmetric(left.right, right.left))

def mirror_tree(node):
    if not node:
        return None
    
    # Create a new mirrored node
    mirrored = TreeNode(node.val)
    mirrored.left = mirror_tree(node.right)
    mirrored.right = mirror_tree(node.left)
    
    return mirrored